Eventy/src/main/java/org/example/eventy/solutions/controllers/ServiceController.java

```diff
@@ -19,6 +23,51 @@ public class ServiceController {
     @Autowired
     private ServiceService serviceService;
 
+    @PostMapping(consumes = MediaType.APPLICATION_JSON_VALUE, produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<CreatedServiceDTO> createService(@RequestBody CreateServiceDTO service) {
+        CreatedServiceDTO response = serviceService.createService(service);
+
+        return new ResponseEntity<CreatedServiceDTO>(response, HttpStatus.CREATED);
+    }
+
+    // NOTE: Add extra RequestParams as necessary if more filters are needed
+    /*@GetMapping(produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<Collection<GetServiceDTO>> getServices(@RequestParam (required = false) String name,
+                                                            @RequestParam (required = false) CategoryDTO category,
+                                                            @RequestParam (required = false) EventTypeDTO eventType,
+                                                            @RequestParam (required = false) double minPrice,
+                                                            @RequestParam (required = false) double maxPrice,
+                                                            @RequestParam (required = false) boolean available) {
+        Collection<GetServiceDTO> services = serviceService.getServices(name, category, eventType, minPrice, maxPrice, available);
+
+        return new ResponseEntity<>(services, HttpStatus.OK);
+    }*/
+
+    @GetMapping(value = "/{id}", produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<GetServiceDTO> getService(@PathVariable("id") long id) {
+        Optional<GetServiceDTO> service = serviceService.getService(id);
+        if (!service.isPresent()) {
+            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+        }
+        return new ResponseEntity<>(service.get(), HttpStatus.OK);
+    }
+
+    @PutMapping(value = "/{id}", consumes = MediaType.APPLICATION_JSON_VALUE, produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<UpdatedServiceDTO> updateService(@RequestBody UpdateServiceDTO service, @PathVariable("id") long id) {
+        service.setId(id);
+        Optional<UpdatedServiceDTO> updatedService = serviceService.updateService(service);
+        if (!updatedService.isPresent()) {
+            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
+        }
+        return new ResponseEntity<>(updatedService.get(), HttpStatus.OK);
+    }
+
+    @DeleteMapping(value = "/{id}")
```

Comment:
Should this be logic delete or true delete? And if it is logic, should it be done with @DeleteMapping? Just asking questions that may be relevant, if not you can completely ignore this haha